packages/@react-spectrum/overlays/src/Popover.tsx

```diff
@@ -63,13 +70,83 @@ function Popover(props: PopoverProps, ref: RefObject<HTMLDivElement>) {
         <button {...dismissButtonProps} />
       </VisuallyHidden>
       {children}
-      {hideArrow ? null : <div className={classNames(styles, 'spectrum-Popover-tip')} {...arrowProps} data-testid="tip" />}
+      {hideArrow ? null : (
+        <Arrow arrowProps={arrowProps} direction={arrowPlacement[placement.split(' ')[0]]} />
+      )}
       <VisuallyHidden>
         <button {...dismissButtonProps} />
       </VisuallyHidden>
     </div>
   );
 }
 
+let ROOT_2 = Math.sqrt(2);
+
+function Arrow(props) {
```

Comment:
We should move this to its own file, and include a link to the original library (and probably its license?). @dannify is working on how to do that.